server/src/main/java/org/apache/druid/server/coordinator/CoordinatorDynamicConfig.java

```diff
@@ -126,6 +129,15 @@ public CoordinatorDynamicConfig(
     this.mergeSegmentsLimit = mergeSegmentsLimit;
     this.maxSegmentsToMove = maxSegmentsToMove;
 
+    if (percentOfSegmentsToConsiderPerMove == null) {
+      log.warn("percentOfSegmentsToConsiderPerMove was null! This is likely because your metastore does not "
```

Comment:
I don't think this should be a warn log message. A user should not be expected to set an optional config. If the config is not set, it seems reasonable to me that the default is set to preserve the old behavior (ie all segments are considered per move). 

For users who want to use the new behavior, I suspect they will hear about it in the release notes or the docs, and update the config to get the new functionality.
 
```suggestion
      log.debug("percentOfSegmentsToConsiderPerMove was null! This is likely because your metastore does not "
```

Reply:
ya, I guess this is fine as a debug. It will actually automatically get set to the default (or the user specified value) the next time a user submits a new dynamic config since that code path uses builder which will populate missing values. will update now